clients/venice-thin-client/src/test/java/com/linkedin/venice/client/stats/BasicClientStatsTest.java

```diff
@@ -27,12 +65,228 @@ public void testMetricPrefix() {
     String prefix = "test_prefix";
     MetricsRepository metricsRepository2 = new MetricsRepository();
     ClientConfig config2 = new ClientConfig(storeName).setStatsPrefix(prefix);
-    BasicClientStats.getClientStats(metricsRepository2, storeName, RequestType.SINGLE_GET, config2);
+    BasicClientStats.getClientStats(metricsRepository2, storeName, SINGLE_GET, config2, FAST_CLIENT);
     // Check metric name
     assertTrue(metricsRepository2.metrics().size() > 0);
     String metricPrefix2 = "." + prefix + "_" + storeName;
     metricsRepository2.metrics().forEach((k, v) -> {
       assertTrue(k.startsWith(metricPrefix2));
     });
   }
+
+  @Test
+  public void testEmitHealthyMetrics() {
+    String storeName = "test_store";
+    String otelPrefix = "test_prefix";
+    InMemoryMetricReader inMemoryMetricReader = InMemoryMetricReader.create();
+    VeniceMetricsConfig metricsConfig = new VeniceMetricsConfig.Builder().setEmitOtelMetrics(true)
+        .setMetricPrefix(otelPrefix)
+        .setOtelAdditionalMetricsReader(inMemoryMetricReader)
+        .setMetricEntities(CLIENT_METRIC_ENTITIES)
+        .setTehutiMetricConfig(new MetricConfig())
+        .build();
+    VeniceMetricsRepository metricsRepository = new VeniceMetricsRepository(metricsConfig);
+    BasicClientStats stats = BasicClientStats
+        .getClientStats(metricsRepository, storeName, SINGLE_GET, new ClientConfig(storeName), THIN_CLIENT);
+
+    String metricPrefix = "." + storeName;
+
+    int httpStatus = SC_OK;
+    int keyCount = 2;
+    double latency = 90.0;
+
+    stats.emitHealthyRequestMetrics(latency, keyCount, httpStatus);
+
+    // validate tehuti metrics
+    Map<String, ? extends Metric> metrics = metricsRepository.metrics();
+    Metric requestMetric = metrics.get(metricPrefix + "--request.OccurrenceRate");
+    Metric healthyRequestMetric = metrics.get(metricPrefix + "--healthy_request.OccurrenceRate");
+    Metric healthyLatencyMetric = metrics.get(metricPrefix + "--healthy_request_latency.Avg");
+    Metric successKeyCountMetric = metrics.get(metricPrefix + "--success_request_key_count.Avg");
+
+    Assert.assertTrue(requestMetric.value() > 0.0);
+    Assert.assertTrue(healthyRequestMetric.value() > 0.0);
+    Assert.assertEquals(healthyLatencyMetric.value(), 90.0);
+    Assert.assertEquals(successKeyCountMetric.value(), 2.0);
+
+    // validate otel metrics
+    Attributes expectedAttributes = getExpectedAttributes(storeName, httpStatus, VeniceResponseStatusCategory.SUCCESS);
+    Collection<MetricData> metricsData = inMemoryMetricReader.collectAllMetrics();
+    Assert.assertFalse(metricsData.isEmpty(), "Metrics should not be empty");
+    Assert.assertEquals(metricsData.size(), 3, "There should be four metrics recorded");
```

Comment:
typo in comments?

Reply:
fixed a few of these copy paste errors. Thanks for catching.